Falining test after upgrade to Groovy 2.4.3

When upgraded to Groovy 2.4.3 one test starts to fail:
```
com.ofg.infrastructure.web.correlationid.CorrelationIdAspectSpec > should set correlationId on header via aspect in asynchronous call                                                                                        FAILED
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is java.lang.ClassCastException: com.ofg.infrastructure.web.correlationid.CorrelationIdAspectSpec$CorrelationIdAspectSpecConfiguration$AspectTestingController cannot be cast to com.ofg.infrastructure.web.correlationid.CorrelationIdAspectSpec
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:978)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:857)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:842)
	at org.springframework.test.web.servlet.TestDispatcherServlet.service(TestDispatcherServlet.java:65)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.springframework.mock.web.MockFilterChain$ServletFilterProxy.doFilter(MockFilterChain.java:167)
	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
	at com.ofg.infrastructure.web.correlationid.CorrelationIdFilter.doFilterInternal(CorrelationIdFilter.groovy:55)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.springframework.mock.web.MockFilterChain.doFilter(MockFilterChain.java:134)
	at org.springframework.test.web.servlet.MockMvc.perform(MockMvc.java:144)
	at com.ofg.infrastructure.web.correlationid.CorrelationIdAspectSpec.should set correlationId on header via aspect in asynchronous call(CorrelationIdAspectSpec.groovy:60)
Caused by: java.lang.ClassCastException: com.ofg.infrastructure.web.correlationid.CorrelationIdAspectSpec$CorrelationIdAspectSpecConfiguration$AspectTestingController cannot be cast to com.ofg.infrastructure.web.correlationid.CorrelationIdAspectSpec
	at com.ofg.infrastructure.web.correlationid.CorrelationIdAspectSpec$CorrelationIdAspectSpecConfiguration$AspectTestingController.asyncPing_closure1(CorrelationIdAspectSpec.groovy:88)
	at com.ofg.infrastructure.web.correlationid.CorrelationIdAspect.wrapWithCorrelationId_closure1(CorrelationIdAspect.groovy:68)
	at com.ofg.infrastructure.correlationid.CorrelationIdUpdater$1.call(CorrelationIdUpdater.groovy:118)
	at org.springframework.web.context.request.async.WebAsyncManager$4.run(WebAsyncManager.java:316)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.lang.Thread.run(Thread.java:745)
```

--- a/build.gradle
+++ b/build.gradle
@@ -104,7 +104,7 @@ configure(srcSubprojects) {
 
     dependencyManagement {
         dependencies {
-            "org.codehaus.groovy:groovy-all" "2.4.1"
+            "org.codehaus.groovy:groovy-all" "2.4.3"
             "com.google.guava:guava" "18.0"
             "commons-io:commons-io" "2.4"
 

--- a/micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/correlationid/CorrelationIdAspectSpec.groovy
+++ b/micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/correlationid/CorrelationIdAspectSpec.groovy
@@ -12,6 +12,7 @@ import org.junit.contrib.java.lang.system.ProvideSystemProperty
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.boot.test.SpringApplicationContextLoader
 import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Import
 import org.springframework.http.MediaType
 import org.springframework.scheduling.annotation.EnableAsync
 import org.springframework.test.context.ContextConfiguration
@@ -66,38 +67,40 @@ class CorrelationIdAspectSpec extends MicroserviceMvcWiremockSpec {
 
     @CompileStatic
     @Configuration
+    @Import(AspectTestingController)
     @EnableAsync
     static class CorrelationIdAspectSpecConfiguration {
+    }
 
-        @RestController
-        @TypeChecked
-        @PackageScope
-        static class AspectTestingController {
+    @RestController
+    @TypeChecked
+    @PackageScope
+    static class AspectTestingController {
 
-            @Autowired private ServiceRestClient serviceRestClient
-            @Autowired private HttpMockServer httpMockServer
+        @Autowired private ServiceRestClient serviceRestClient
+        @Autowired private HttpMockServer httpMockServer
 
-            @RequestMapping(value = "/syncPing", method = RequestMethod.GET, produces = MediaType.TEXT_PLAIN_VALUE)
-            String syncPing() {
+        @RequestMapping(value = "/syncPing", method = RequestMethod.GET, produces = MediaType.TEXT_PLAIN_VALUE)
+        String syncPing() {
+            callWiremockAndReturnOk()
+        }
+
+        @RequestMapping(value = "/asyncPing", method = RequestMethod.GET, produces = MediaType.TEXT_PLAIN_VALUE)
+        Callable<String> asyncPing() {
+            return {
                 callWiremockAndReturnOk()
             }
+        }
 
-            @RequestMapping(value = "/asyncPing", method = RequestMethod.GET, produces = MediaType.TEXT_PLAIN_VALUE)
-            Callable<String> asyncPing() {
-                return {
-                    callWiremockAndReturnOk()
-                }
-            }
-
-            private String callWiremockAndReturnOk() {
-                serviceRestClient.forExternalService()
-                        .get()
-                        .onUrl(new URI("http://localhost:${httpMockServer.port()}"))
-                        .andExecuteFor()
-                        .aResponseEntity()
-                        .ofType(String)
-                return "OK"
-            }
+        private String callWiremockAndReturnOk() {
+            serviceRestClient.forExternalService()
+                    .get()
+                    .onUrl(new URI("http://localhost:${httpMockServer.port()}"))
+                    .andExecuteFor()
+                    .aResponseEntity()
+                    .ofType(String)
+            return "OK"
         }
     }
+
 }

